packages/gatsby-image/src/index.js

```diff
@@ -141,7 +123,7 @@ class Image extends React.Component {
 
     // If this image has already been loaded before then we can assume it's
     // already in the browser cache so it's cheap to just show directly.
-    const seenBefore = inImageCache(props)
+    const seenBefore = !props.critical && inImageCache(props)
```

Comment:
This may be a bit controvertial - if a critical image is used in multiple places on a page, but has different fadeIn values - one of the critical image may not appear when the component is mounted.